lib/src/fsmonitor.rs

```diff
@@ -44,20 +52,26 @@ pub enum FsmonitorKind {
     None,
 }
 
-impl FromStr for FsmonitorKind {
-    type Err = config::ConfigError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "watchman" => Ok(Self::Watchman),
-            "test" => Err(config::ConfigError::Message(
-                "cannot use test fsmonitor in real repository".to_string(),
-            )),
-            "none" => Ok(Self::None),
-            other => Err(config::ConfigError::Message(format!(
-                "unknown fsmonitor kind: {}",
-                other
-            ))),
+impl FsmonitorSettings {
+    /// Creates an `FsmonitorSettings` from a `config`.
+    pub fn from_config(config: &Config) -> Result<FsmonitorSettings, ConfigError> {
+        match config.get_string("core.fsmonitor") {
+            Ok(s) => match s.as_str() {
+                "watchman" => Ok(Self::Watchman(WatchmanConfig {
+                    register_trigger: config
+                        .get_bool("core.watchman.register_trigger")
+                        .unwrap_or_default(),
```

Comment:
nit: `.optional()?.unwrap_or_default()` can be used to propagate type error (with `use ConfigResultExt`)

Reply:
Done